workspaces/src/network/transaction.rs

```diff
@@ -0,0 +1,323 @@
+use std::convert::TryInto;
+
+use near_crypto::KeyType;
+use near_primitives::transaction::{
+    Action, AddKeyAction, CreateAccountAction, DeleteAccountAction, DeleteKeyAction,
+    DeployContractAction, FunctionCallAction, StakeAction, TransferAction,
+};
+use near_primitives::views::FinalExecutionOutcomeView;
+
+use crate::network::{CallExecution, CallExecutionDetails, Network, ViewResultDetails};
+use crate::rpc::client::{
+    send_batch_tx_and_retry, Client, DEFAULT_CALL_DEPOSIT, DEFAULT_CALL_FN_GAS,
+};
+use crate::types::{AccessKey, AccountId, Balance, Gas, InMemorySigner, PublicKey, SecretKey};
+use crate::worker::Worker;
+use crate::Account;
+
+/// A set of arguments we can provide to a transaction, containing
+/// the function name, arguments, the amount of gas to use and deposit.
+#[derive(Debug, Clone)]
+pub struct Function {
+    name: String,
+    args: Vec<u8>,
+    deposit: Balance,
+    gas: Gas,
+}
+
+impl Function {
+    pub fn new(name: &str) -> Self {
+        Self {
+            name: name.into(),
+            args: vec![],
+            deposit: DEFAULT_CALL_DEPOSIT,
+            gas: DEFAULT_CALL_FN_GAS,
+        }
+    }
+
+    pub fn args(mut self, args: Vec<u8>) -> Self {
+        self.args = args;
+        self
+    }
+
+    pub fn args_json<U: serde::Serialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = serde_json::to_vec(&args)?;
+        Ok(self)
+    }
+
+    pub fn args_borsh<U: borsh::BorshSerialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = args.try_to_vec()?;
+        Ok(self)
+    }
+
+    pub fn deposit(mut self, deposit: u128) -> Self {
+        self.deposit = deposit;
+        self
+    }
+
+    pub fn gas(mut self, gas: u64) -> Self {
+        self.gas = gas;
+        self
+    }
+}
+
+impl From<Function> for Action {
+    fn from(function: Function) -> Self {
+        Self::FunctionCall(FunctionCallAction {
+            method_name: function.name,
+            args: function.args,
+            deposit: function.deposit,
+            gas: function.gas,
+        })
+    }
+}
+
+pub struct Transaction<'a> {
```

Comment:
Can you document this? I think it isn't intuitive the ambiguity mentioned by @miraclx but also that this is a builder pattern that is generated through something else